app/controllers/admin/subscriptions_controller.rb

```diff
@@ -52,10 +52,30 @@ def unlink_course
       redirect_back fallback_location: root_path, notice: 'Cours retiré avec succès !'
     end
 
+    def confirm
+      subscription = Subscription.find(params[:id])
+      if subscription.confirmed?
+        redirect_to admin_subscriptions_path, alert: "L'inscription est deja confirmée !"
+      else
+        subscription.confirmed!
+        redirect_to admin_subscriptions_path, notice: 'Inscription confirmée avec succès !'
+      end
+    end
+
+    def archive
+      subscription = Subscription.find(params[:id])
+      if subscription.archived?
+        redirect_to admin_subscriptions_path, alert: "L'inscription est deja archivée !"
+      else
+        subscription.archived!
+        redirect_to admin_subscriptions_path, notice: 'Inscription archivée avec succès !'
+      end
+    end
+
     private
 
     def subscription_params
-      params.require(:subscription).permit(:member_id, course_ids: [])
+      params.require(:subscription).permit(:member_id, :status, course_ids: [])
```

Comment:
Unless you want to add a `select` tag in the `admin/subscriptions` form to allow the admins to change a subscription's status from there, this is not needed